### ISSUE TITLE
Running not neural network framework workloads

### ISSUE THREAD

**Issue by pranasblk** [kubeflow/katib#706]
It would not be nice to use framework for custom implementations (not only well known neural network frameworks). Thanks for consideration.

**Describe the solution you'd like**
Specify kind of job custom and write results to the form framework understands e.g. pipe custom output to formatting program specific to katib.

**Comment by johnugeorge** [comment]
Can you explain your needs in more detail?

**Comment by pranasblk** [comment]
Hi George,

I'd love to run hyper parameters optimisation of JVM process with specific
fitness functions e.g. Sortino for automated trading strategy replaying
historical/simulated input data on Kubernetes cluster.

Let me know what you think about using existing toolset outside of
ANN/Python environments.

Thanks

**Comment by johnugeorge** [comment]
It does not depend on ANNs. You could provide container image and command in https://github.com/kubeflow/katib/blob/master/examples/v1alpha2/random-example.yaml#L20 I haven't tried it though, it should work. 
The only requirement is that container should emit metric (stdout) in metric_name=metric_value (eg: https://github.com/kubeflow/pytorch-operator/blob/master/examples/mnist/mnist.py#L64)
